library/ceph_disks.py

```diff
@@ -0,0 +1,133 @@
+#!/usr/bin/python
+#coding: utf-8 -*-
+
+DOCUMENTATION = '''
+---
+author: Siva Mullapudi
+module: ceph_disks
+short_description: Outputs a list of valid devices
+description:
+    - Outputs a list of devices based on the osd type.
+options: none
+'''
+
+EXAMPLES = '''
+# Assuming a journal_collocation osd type
+- ceph_disks:
+  register: ssds
+'''
+
+import os
+import re
+import subprocess
+import json
+from subprocess import Popen, PIPE
+
+def run_command(cmd):
+    process = Popen(cmd, stdout=PIPE, stderr=PIPE, shell=True)
+    stdout, stderr = process.communicate()
+    return stdout.rstrip('\n')
+
+def has_valid_mountpoint(device_name, lsblk_devices):
+    partition_pattern = re.compile(device_name + "[0-9]")
+
+    for device in lsblk_devices.splitlines():
+        if partition_pattern.match(device.split()[0]):
+            if len(device.split()) >= 7:
+                mountPoint = device.split()[6]
+                if mountPoint == "/boot" or mountPoint == "/" or mountPoint == "[SWAP]":
+                    return False
+
+    return True
+
+def disk_type(device_name, device_size, adaptecTest, lsiTest, path):
+    #ADAPTEC CONTROLLER
+    if adaptecTest:
+        temp = run_command("/lib/udev/scsi_id --whitelist -p 0x80 --device=/dev/" + device_name + " --export | grep ID_SERIAL_SHORT")
+        serial = temp.rpartition('=')[0]
+
+        if serial:
+            temp = run_command(path + "/arcconf getconfig 1 LD | grep -A20 \"" + serial + "\" | grep \"Segment 0\"")
+            longSerial = temp[-15:]
+            isSSD = run_command(path + "/arcconf getconfig 1 PD | grep -A13 " + longSerial + " | grep SSD | grep Yes")
```

Comment:
Can we move the grepping into native python text searching on the results from the command? We should keep the shelling out to things that can't be done in python. In this case, running `scsi_id` or `arcconf`. The rest should be doable as native Python methods.